core/che-core-api-core/src/main/java/org/eclipse/che/api/core/Page.java

```diff
@@ -0,0 +1,312 @@
+     * to the value returned by the {@link #getItemsCount()} method.
+     */
+    public int getSize() {
+        return pageSize;
+    }
+
+    /**
+     * Returns page number starting from 1.
+     *
+     * <p>If the page is not regular page(it refers rather to the
+     * data window than to the certain page(e.g. skip=2, pageSize=4))
+     * then this method returns -1.
+     */
+    public long getNumber() {
+        if (itemsBefore % pageSize != 0) {
+            return -1;
+        }
+        return itemsBefore / pageSize + 1;
+    }
+
+    /**
+     * Returns the size of the page items, returned value
+     * may be equal to 0 when the page {@link #isEmpty() is empty},
+     * the values is the same to {@code page.getItems().size()}.
+     */
+    public int getItemsCount() {
+        return items.size();
+    }
+
+    /** Returns the count of all the items. */
+    public long getTotalItemsCount() {
+        return totalCount;
+    }
+
+    /**
+     * Returns page items or an empty list when page doesn't contain items.
+     *
+     * <p>Note that returned instance is modifiable list and modification applied
+     * on that list will affect the origin page result items, which allows
+     * components to modify items before propagating page.
+     */
+    public List<ITEM_T> getItems() {
+        return items;
+    }
+
+    /**
+     * Gets the page items and maps them with given {@code mapper}.
+     *
+     * @param mapper
+     *         items mapper
+     * @param <R>
+     *         the type of the result items
+     * @return the list of mapped items
+     */
+    public <R> List<R> getItems(Function<? super ITEM_T, ? extends R> mapper) {
+        requireNonNull(mapper, "Required non-null mapper for page items");
+        return items.stream()
+                    .map(mapper::apply)
+                    .collect(toList());
+    }
+
+    /**
+     * Fills the given collection with page items.
+     * This method may be convenient when needed collection
+     * different from the {@link List}.
+     *
+     * <p>The common example:
+     * <pre>{@code
+     *      Set<User> user = page.fill(new TreeSet<>(comparator));
+     * }</pre>
+     *
+     * <p>Note that this method uses {@code container.addAll(items)}
+     * so be aware of putting modifiable collection.
+     *
+     * @param container
+     *         collection which is used to fill result into
+     * @param <COL_T>
+     *         collection type
+     * @return given collection instance {@code container} with items filled
+     * @throws NullPointerException
+     *         when {@code container} is null
+     */
+    public <COL_T extends Collection<? super ITEM_T>> COL_T fill(COL_T container) {
+        requireNonNull(container, "Required non-null items container");
+        container.addAll(items);
+        return container;
+    }
+
+    /** Represents page reference as a combination of {@code skipItems & pageSize}. */
+    public static class PageRef {
```

Comment:
Because it is strongly related to the `Page` + there is no reason to create `PageRef` from the other place.